core/storage/trie/serialization/trie_serializer_impl.cpp

```diff
@@ -55,39 +56,44 @@ namespace kagome::storage::trie {
   outcome::result<RootHash> TrieSerializerImpl::storeRootNode(
       TrieNode &node, StateVersion version) {
     auto batch = backend_->batch();
+    BOOST_ASSERT(batch != nullptr);
 
-    OUTCOME_TRY(enc,
-                codec_->encodeNode(node,
-                                   version,
-                                   [&](const TrieNode *,
-                                       common::BufferView hash,
-                                       common::Buffer &&encoded) {
-                                     return batch->put(hash,
-                                                       std::move(encoded));
-                                   }));
-    auto key = codec_->hash256(enc);
-    OUTCOME_TRY(batch->put(key, std::move(enc)));
+    OUTCOME_TRY(
+        enc,
+        codec_->encodeNode(node,
+                           version,
+                           [&](TrieNode const &node,
+                               common::BufferView hash,
+                               common::Buffer &&encoded) {
+                             current_stats_.new_nodes_written++;
+                             if (node.getValue()) {
+                               current_stats_.values_written++;
+                             }
```

Comment:
Did you mean "v1 hashed values written"?
```suggestion
                             if (node) {
                               current_stats_.new_nodes_written++;
                             } else {
                               current_stats_.values_written++;
                             }
```
Or do you want to store hashed values manually?
```suggestion
                             current_stats_.new_nodes_written++;
                             if (node.getValue().hash) {
                               current_stats_.values_written++;
                               OUTCOME_TRY(batch->put(node.getValue().hash, node.getValue().value));
                             }
```